log/scope.go

```diff
@@ -130,27 +165,33 @@ func (s *Scope) FatalEnabled() bool {
 }
 
 // Error outputs a message at error level.
-func (s *Scope) Error(msg string, fields ...zapcore.Field) {
+//func (s *Scope) Error(msg string, fields ...zapcore.Field) {
+func (s *Scope) Error(fields ...interface{}) {
 	if s.GetOutputLevel() >= ErrorLevel {
-		s.emit(zapcore.ErrorLevel, s.GetStackTraceLevel() >= ErrorLevel, msg, fields)
+		ie, firstIdx := getErrorStruct(fields)
+		s.callHandlers(ErrorLevel, s, ie, fields[0].(string), toZapSlice(firstIdx+1, fields))
 	}
 }
 
 // Errora uses fmt.Sprint to construct and log a message at error level.
+//func (s *Scope) Errora(args ...interface{}) {
 func (s *Scope) Errora(args ...interface{}) {
 	if s.GetOutputLevel() >= ErrorLevel {
-		s.emit(zapcore.ErrorLevel, s.GetStackTraceLevel() >= ErrorLevel, fmt.Sprint(args...), nil)
+		ie, firstIdx := getErrorStruct(args)
+		s.callHandlers(ErrorLevel, s, ie, fmt.Sprint(args[firstIdx:]...), nil)
 	}
 }
 
 // Errorf uses fmt.Sprintf to construct and log a message at error level.
-func (s *Scope) Errorf(template string, args ...interface{}) {
+// func (s *Scope) Errorf(template string, args ...interface{}) {
+func (s *Scope) Errorf(args ...interface{}) {
```

Comment:
This *might* be a backwards incompatible API change. Or maybe not. But why do we need it? Is it valid to pass `Errorf(<not string>, something)`? What happens in this case?

Reply:
we need it to retain the familiar interface to scope.Debug* et al. while allowing structured.Error to be optionally passed. the alternative (per thread with doug) is to call the functions taking structured.Error something else. but what? i feel it could get messy to have a proliferation of these functions with slightly different names.